renderer.js

```diff
@@ -0,0 +1,293 @@
+                    store.dispatch({
+                        type: types["DURING_PAGE_FADE_IN"]
+                    });
+
+                    // show previous step
+                    $('#f fieldset:nth-child(' + store.getState().nextPage + ')').fadeIn(400, function() {
+                        store.dispatch({
+                            type: types["MOVE_PAGE_COMPLETED"]
+                        });
+                    });
+
+                    // scroll window to beginning of the form
+                    scrollToClass( $('html'), 20 );
+                });
+            }
+            if(state.nextPage == 1) {
+                $('.btn-prev, .btn-submit').hide();
+                $('.btn-next').show();
+            }
+            else if(state.nextPage == 7) {
+                $('.btn-next').hide();
+                $('.btn-prev, .btn-submit').show();
+            }
+            else {
+                $('.btn-prev, .btn-next').show();
+                $('.btn-submit').hide();
+            }
+            updateProgress(state.nextPage);
+        }
+    }
+
+    state = store.getState();
+
+    if(state.toggleNextBtn !== null) {
+        $('.btn-next').toggleClass('disabled', !state.toggleNextBtn);
+        store.dispatch({
+            type: types["TOGGLE_NEXT_BUTTON_COMPLETED"]
+        });
+    }
+
+    state = store.getState();
+
+    if(state.validatePage) {
+        var bool;
+        switch(state.currentPage) {
+            case 1:
+                bool = state.isProdVerValid && state.productType !== null;
+                break;
+            // There is no case 2: 
+            case 3:
+                bool = state.isProbURLValid && (!state.isBrowserOther || !!$("#browser-name input").val()); //may need to validate browser name as well
+                break;
+            case 5:
+                bool = state.isScreenshotURLValid;
+                break;
+            default:
+                bool = true;
+        }
+        store.dispatch({
+            type: types["PAGE_FORM_VALIDATE"],
+            data: bool
+        });
+    }
+
+    state = store.getState();
+
+    if(state.updateChecklistSkip) {
+        $('.checklist').each(function() {
+            var skip = $(this).data('skip');
+            $(this).hide().toggleClass('skip', shouldSkip(skip, state.productType, state.problemType));
+        });
+        store.dispatch({
+            type: types["UPDATE_CHECKLIST_SKIPPED_ENTRY_COMPLETED"]
+        });
+    }
+
+    state = store.getState();
+
+    if(state.showFirstChecklist) {
+        $('.checklist, .checklist ~ .resolved').hide();
+        $('.checklist label.btn').removeClass('active').find('input').prop('checked', false);
+        $('.checklist').eq(0).fadeIn();
+        store.dispatch({
+            type: types["SHOW_CHECKLIST_COMPLETE"],
+            data: 0
+        });
+    }
+
+    state = store.getState();
+
+    if(state.showNextChecklist) {
+        $('.checklist ~ .resolved').hide();
```

Comment:
Why you don't using React?
Same template logic with declarative rules will be more more clear than spaghetti-like code with many show()/hide() calls.
This code is not refactorable because it's cyclomatic complexity too high.